src/utility_belt/sql/conv.clj

```diff
@@ -60,6 +60,11 @@
     (.setObject ^PreparedStatement statement idx
                 (value-to-json-pgobject value)))
   IPersistentVector
-  (set-parameter [value statement idx]
-    (.setObject ^PreparedStatement statement idx
-                (value-to-json-pgobject value))))
+  (set-parameter [v ^PreparedStatement s ^long i]
+   (let [conn (.getConnection s)
+          meta (.getParameterMetaData s)
+          type-name (.getParameterTypeName meta i)]
+     (if-let [elem-type (when type-name (second (re-find #"^_(.*)" type-name)))]
+       (.setObject s i (.createArrayOf conn elem-type (to-array v)))
```

Comment:
Check this out: https://github.com/seancorfield/next-jdbc/blob/develop/src/next/jdbc/types.clj - next.jdbc has a shortcut ns for turning things into types

Reply:
@lukaszkorecki  I tried to use it but it doesn't solve the issue I was having (well I can't get the tests to pass in lucien using that ns). Sometimes we want the things to be turned into json and sometimes we want vector of uuids etc.

Reply:
Right, moving on then.